File: authserver/src/main/java/l2s/authserver/network/gamecomm/gs2as/PlayerInGame.java
package l2s.authserver.network.gamecomm.gs2as;

import io.netty.buffer.ByteBuf;
import l2s.authserver.network.gamecomm.ReceivablePacket;
import l2s.authserver.network.gamecomm.vertx.GameServerConnection;

public class PlayerInGame extends ReceivablePacket {
    private final String account;

    public PlayerInGame(GameServerConnection client, ByteBuf byteBuf) {
        super(client, byteBuf);

        account = readString(byteBuf);
    }

    @Override
    protected void runImpl(GameServerConnection client) {
        if (client.isAuthed())
            client.getGameServerDescription().addAccount(account);
    }
}

File: authserver/src/main/java/l2s/authserver/network/gamecomm/gs2as/PlayerInGame.kt
package l2s.authserver.network.gamecomm.gs2as

import io.netty.buffer.ByteBuf
import l2s.authserver.network.gamecomm.GameServerDescription
import l2s.authserver.network.gamecomm.ReceivablePacket
import l2s.authserver.network.gamecomm.vertx.GameServerConnection

class PlayerInGame(client: GameServerConnection, byteBuf: ByteBuf) : ReceivablePacket(client, byteBuf) {

    private val account: String = readString(byteBuf)

    override fun runImpl(client: GameServerConnection) {
        val gs: GameServerDescription = client.description
        if (client.isAuthed)
            gs.addAccount(account)
    }

}